src/PortingAssistant.Client.Analysis/AnalysisHandler.cs

```diff
@@ -384,6 +395,51 @@ private List<ProjectResult> AnalyzeActions(List<string> projects, string targetF
             }
         }
 
+        private ProjectCompatibilityResult GenerateCompatibilityResults(List<SourceFileAnalysisResult> sourceFileAnalysisResults, string projectPath, bool isPorted)
```

Comment:
Can this be method be improved runtime wise? I do see loops inside loops...

Reply:
I think that is the analyze result data structure look like, we have to loop files to get all api results.

Reply:
Yep. we need to refactor this class at some point to make the api compatibility clearer, will add an issue for that.